main/src/ca/uwaterloo/flix/language/ast/SyntaxTree.scala

```diff
@@ -0,0 +1,489 @@
+      case object LiteralArray extends Expr
+
+      case object LiteralList extends Expr
+
+      case object LiteralMap extends Expr
+
+      case object LiteralMapKeyValueFragment extends Expr
+
+      case object LiteralRecord extends Expr
+
+      case object LiteralRecordFieldFragment extends Expr
+
+      case object LiteralSet extends Expr
+
+      case object LiteralVector extends Expr
+
+      case object Match extends Expr
+
+      case object MatchRuleFragment extends Expr
+
+      case object NewObject extends Expr
+
+      case object OpenVariant extends Expr
+
+      case object OpenVariantAs extends Expr
+
+      case object Paren extends Expr
+
+      case object ParYield extends Expr
+
+      case object ParYieldFragment extends Expr
+
+      case object RecordOperation extends Expr
+
+      case object RecordOpExtend extends Expr
+
+      case object RecordOpRestrict extends Expr
+
+      case object RecordOpUpdate extends Expr
+
+      case object RecordSelect extends Expr
+
+      case object Ref extends Expr
+
+      case object Scope extends Expr
+
+      case object ScopeName extends Expr
+
+      case object Spawn extends Expr
+
+      case object Statement extends Expr
+
+      case object Static extends Expr
+
+      case object StringInterpolation extends Expr
+
+      case object Try extends Expr
+
+      case object TryCatchBodyFragment extends Expr
+
+      case object TryCatchRuleFragment extends Expr
+
+      case object TryWithBodyFragment extends Expr
+
+      case object TryWithRuleFragment extends Expr
+
+      case object Tuple extends Expr
+
+      case object TypeMatch extends Expr
+
+      case object TypeMatchRuleFragment extends Expr
+
+      case object Unary extends Expr
+
+      case object UncheckedCast extends Expr
+
+      case object UncheckedMaskingCast extends Expr
+
+      case object Use extends Expr
+
+      case object Without extends Expr
+
+    }
+
+    //////////////////////////////////////////////////////////////////////////////////////////
+    /// TYPES ////////////////////////////////////////////////////////////////////////////////
+    //////////////////////////////////////////////////////////////////////////////////////////
+    sealed trait Type extends TreeKind
+
+    object Type {
+      /**
+       * A marker kind used to wrap nested types.
+       * For instance on a tuple type "(Int32, Bool)" you would do
+       * Type
+       * Tuple
+       * Type
+       * Ident
+       * Type
+       * Ident
+       */
+      case object Type extends Type
+
+      case object Apply extends Type
+
+      case object Argument extends Type
+
+      case object ArgumentList extends Type
+
+      case object Ascribe extends Type
+
+      case object Binary extends Type
+
+      case object CaseSet extends Type
+
+      case object Constant extends Type
+
+      case object Constraint extends Type
+
+      case object ConstraintList extends Type
+
+      case object DerivationList extends Type
```

Comment:
I am not sure if this belongs under type. It is really only for enums, right? So maybe it goes at the top in the unsorted bucket?

Reply:
Yea that's only for enums (and restrictable enums). I will move it to the top-most namespace